src/MongoDbEventStoreAdapter.php

```diff
@@ -0,0 +1,218 @@
+    protected $standardColumns = ['event_id', 'event_name', 'event_class', 'created_at', 'payload', 'version'];
+
+    /**
+     * @param  array $configuration
+     * @throws \Prooph\EventStore\Adapter\Exception\ConfigurationException
+     */
+    public function __construct(array $configuration)
+    {
+        if (!isset($configuration['mongo_client'])) {
+            throw new ConfigurationException('Mongo client configuration is missing');
+        }
+
+        if (!isset($configuration['db_name'])) {
+            throw new ConfigurationException('Mongo database name is missing');
+        }
+
+        if (isset($configuration['stream_table_map'])) {
+            $this->streamTableMap = $configuration['stream_table_map'];
+        }
+
+        $this->mongoClient = $configuration['mongo_client'];
+        $this->dbName      = $configuration['db_name'];
+
+        if (isset($configuration['serializer_adapter'])) {
+            $this->serializerAdapter = $configuration['serializer_adapter'];
+        }
+    }
+
+    /**
+     * @param Stream $stream
+     * @throws \Prooph\EventStore\Exception\RuntimeException If creation of stream fails
+     * @return void
+     */
+    public function create(Stream $stream)
+    {
+        if (count($stream->streamEvents()) === 0) {
+            throw new RuntimeException(
+                sprintf(
+                    "Cannot create empty stream %s. %s requires at least one event to extract metadata information",
+                    $stream->streamName()->toString(),
+                    __CLASS__
+                )
+            );
+        }
+
+        $this->appendTo($stream->streamName(), $stream->streamEvents());
+    }
+
+    /**
+     * @param StreamName $streamName
+     * @param DomainEvent[] $streamEvents
+     * @throws \Prooph\EventStore\Exception\StreamNotFoundException If stream does not exist
+     * @return void
+     */
+    public function appendTo(StreamName $streamName, array $streamEvents)
+    {
+        foreach ($streamEvents as $event) {
+            $this->insertEvent($streamName, $event);
+        }
+    }
+
+    /**
+     * @param StreamName $streamName
+     * @param null|int $minVersion
+     * @return Stream|null
+     */
+    public function load(StreamName $streamName, $minVersion = null)
+    {
+        $events = $this->loadEventsByMetadataFrom($streamName, array(), $minVersion);
+
+        return new Stream($streamName, $events);
+    }
+
+    /**
+     * @param StreamName $streamName
+     * @param array $metadata
+     * @param null|int $minVersion
+     * @return DomainEvent[]
+     */
+    public function loadEventsByMetadataFrom(StreamName $streamName, array $metadata, $minVersion = null)
+    {
+        $collection = $this->mongoClient->selectCollection($this->dbName, $this->getCollection($streamName));
+
+        if (null !== $minVersion) {
+            $metadata['version'] = ['$gt' => [$minVersion]];
+        }
+
+        $results = $collection->find($metadata);
```

Comment:
Does mongoDb guarantee the same document order as they were inserted? Otherwise, we need to order the result by `version`. When reconstituting ARs it is important that the AR receives the history events in the correct order.